packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -726,8 +873,12 @@ class MultiFrameImageStreamCompleter extends ImageStreamCompleter {
   }
 
   Future<void> _decodeNextFrameAndSchedule() async {
+    _nextFrame?.image.dispose();
+    _nextFrame = null;
     try {
       _nextFrame = await _codec!.getNextFrame();
+      // No need to clone() the image in the next frame since we will not retain
+      // it once we have given it to someone else.
```

Comment:
I don't understand... we seem to be disposing it a few lines above. Aren't we disposing somebody else's handle if we don't retain?

Reply:
Only if it's not null above. Added a comment to clarify.

Reply:
Oh, this comment is just wrong. Removing it. I thought at one point it would work this way but it did not in fact work. I just forgot to remove the comment.